python/GangaDirac/Lib/Files/DiracFile.py

```diff
@@ -137,7 +137,7 @@ def locations(self):
 
     def _setLFNnamePattern(self, lfn="", namePattern=""):
 
-        if self.defaultSE != "":
+        if hasattr(self, 'defaultSE') and self.defaultSE != "":
```

Comment:
The object has a defined slot in the class as per `Objects.py`.
The attribute is only correctly set via the class calling `__init__` and initializing the attributes to the default value.
(I don't know if hasattr will pass when only the slot has been assigned in the metaclass)

However, when a new class is constructed via `__new__` the class doesn't fully assign the attributes when constructing the object as they're not required.

Users are seeing:
```
ERROR    XML from-file error for file:
Could not find attribute defaultSE in DiracFile(namePattern='', lfn='', localDir='None')
```

When attempting to load old jobs which don't have this attribute stored in the XML.

By checking for the attribute before calling it I think this should be enough to fix this bug and make old jobs accessible again.

Wrt the order of objects in the Schema I think the order that objects are constructed within the `Objects.py` is such that the `defaultSE` is assigned to the class first before the other attributes look for it, although I could be wrong.

Ideally things like this would go away if we made Schema objects handled via a defaultdict.